src/date-picker/date-picker.jsx

```diff
@@ -212,9 +212,20 @@ const DatePicker = React.createClass({
    * Open the date-picker dialog programmatically from a parent.
    */
   openDialog() {
-    this.setState({
-      dialogDate: this.getDate(),
-    }, this.refs.dialogWindow.show);
+    /**
+     * if the date is not selected then set it to new date
+     * (get the current system date while doing so)
+     * else set it to the currently selected date
+     */
+    if (this.state.date !== undefined) {
+      this.setState({
+        dialogDate: this.getDate(),
+      }, this.refs.dialogWindow.show);
+    } else {
+      this.setState({
+        dialogDate: new Date(),
```

Comment:
@nathanmarks : good point. I think we can remove it. Let me do the changes and check